services/semaphore.js

```diff
@@ -7,20 +8,41 @@ module.exports = {
 		return Boolean(env.SEMAPHORE);
 	},
 	configuration({env, cwd}) {
-		const pr = env.PULL_REQUEST_NUMBER;
-		const isPr = Boolean(pr);
+		if (env.BRANCH_NAME) {
+			// Exists in v1 only
+			const pr = env.PULL_REQUEST_NUMBER;
+			const isPr = Boolean(pr);
+			return {
```

Comment:
Do not duplicate this whole block. Instead use `||` or ternary if necessary for the properties that varies between v1 and v2